demisto_sdk/commands/common/logger.py

```diff
@@ -101,10 +104,168 @@ def set_demisto_logger(demisto_logger: logging.Logger):
+    if hasattr(logging.getLoggerClass(), method_name):
+        raise AttributeError(f"{method_name} already defined in logger class")
+
+    # This method was inspired by the answers to Stack Overflow post
+    # http://stackoverflow.com/q/2183233/2988730, especially
+    # http://stackoverflow.com/a/13638084/2988730
+    def logForLevel(self, message, *args, **kwargs):
+        if self.isEnabledFor(level_num):
+            self._log(level_num, message, args, **kwargs)
+
+    def logToRoot(message, *args, **kwargs):
+        logging.log(level_num, message, *args, **kwargs)
+
+    logging.addLevelName(level_num, level_name)
+    setattr(logging, level_name, level_num)
+    setattr(logging.getLoggerClass(), method_name, logForLevel)
+    setattr(logging, method_name, logToRoot)
+
+
+SUCCESS_LEVEL: int = 25
+if not hasattr(logging.getLoggerClass(), "success"):
+    _add_logging_level("SUCCESS", SUCCESS_LEVEL)
+
+
+class ColorConsoleFormatter(logging.Formatter):
+    FORMATS = {
+        logging.DEBUG: "[lightgrey]%(message)s[/lightgrey]",
+        logging.INFO: "[lightgrey]%(message)s[/lightgrey]",
+        logging.WARNING: "[yellow][%(levelname)s] %(message)s[/yellow]",
+        logging.ERROR: "[red][%(levelname)s] %(message)s[/red]",
+        logging.CRITICAL: "[red][bold][%(levelname)s] %(message)s[/bold[/red]",
+        SUCCESS_LEVEL: "[green]%(message)s[/green]",
+    }
+
+    def __init__(
+        self,
+    ):
+        super().__init__(
+            fmt="[%(levelname)s] %(message)s",
+            datefmt=DATE_FORMAT,
+        )
+
+    @staticmethod
+    def _record_contains_escapes(record: logging.LogRecord) -> bool:
+        message = record.getMessage()
+        for key in escapes:
+            if not key.startswith("[/]") and key in message:
+                return True
+        return False
+
+    @staticmethod
+    def _string_starts_with_escapes(string: str) -> bool:
+        message = string.strip()
+        return message.startswith("[")
+
+    @staticmethod
+    def _record_starts_with_escapes(record: logging.LogRecord) -> bool:
+        message = record.getMessage().strip()
+        return message.startswith("[")
+
+    @staticmethod
+    def _get_start_escapes(record: logging.LogRecord) -> str:
+        ret_value = ""
+
+        current_message = record.getMessage()
+        while ColorConsoleFormatter._string_starts_with_escapes(current_message):
+
+            if not ColorConsoleFormatter._string_starts_with_escapes(current_message):
+                return ret_value
+
+            # Record starts with escapes - Extract them
+            current_escape = current_message[0 : current_message.find("]") + 1]
+            ret_value += current_escape
+            current_message = current_message[
+                len(current_escape) : current_message.find("]", len(current_escape)) + 1
+            ]
+
+        return ret_value
+
+    @staticmethod
+    def _insert_into_escapes(record: logging.LogRecord, string: str) -> str:
```

Comment:
Can you explain why we need it and what that does?

Reply:
To insert strings inside escaped messages, like when adding the logging level.